### DISCUSSION TITLE
Find and remove the text

### DISCUSSION THREAD

**Discussion by MallRoy** [pymupdf/PyMuPDF#2151] [Looking for help] [answered]
I am using ``` page.get_text("words", sort=True) ``` to return the text contents in the page. I want to remove some of the text on a page that matches the specific coordinates returned from the above snippet

**Comment by JorjMcKie** [comment, marked as answer]
Best use Redaction annotations:
```python
words = page.get_text("words")
page.add_redact_annot(w[:4])  # for some desired w in words
page.add_redact_annot(v[:4])  # for another desired v in words
...
page.apply_redactions()  # remove all text in above redactions
```

**Reply by MallRoy** [reply]
Is there any other work around without using redaction

**Reply by JorjMcKie** [reply]
No, there is not.
What's wrong with redactions?

**Reply by MallRoy** [reply]
A few pdf we handle are corrupted, when we load them with pymupdf and process and which leads to some adverse effects. so we do the additional process for repairing before doing the redaction.
